### PR TITLE
explore: real-time testing

### PR DESCRIPTION
**Pull request by cys1102**
explore:

- Added RealTimeTest class
- Implemented real-time test with sparse flow (previous approach)
- Added saving method for results in internal storage every 100 times

### CODE REVIEW

File: ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    public void setNumOfTry(int numOfTry) { this.numOfTry = numOfTry; }
+    public int getNumOfCorrect() { return numOfCorrect; }
+    public void setNumOfCorrect(int numOfCorrect) { this.numOfCorrect = numOfCorrect; }
+    public float getAccuracy() { return accuracy; }
+    public void setAccuracy() { this.accuracy = (float) (this.numOfCorrect * 100) / this.numOfTry; }
+
+    public void saveResult(Context context) {
+        String TAG = "RealTimeTest";
+        try {
+            FileWriter out = new FileWriter(new File(context.getFilesDir(), "testResult.txt"));
```
Comment (Kalaiz) [inline ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java]: context.[getExternalFilesDir()](https://developer.android.com/training/data-storage/app-specific#external-access-files )

**Review by Kalaiz — CHANGES_REQUESTED** [review]
Looks good to me!
Only issue is that it seems that the internal files can be accessed only [on emulators (test will be limited to just the Computer iin this case ) and rooted devices](https://stackoverflow.com/a/54404504/11200630) and I believe within the app during run time.
https://stackoverflow.com/a/38623141/11200630

There are various workaround for this, but majority of them seems tedious. 
I believe the easiest way would be to write it in an external directory instead of an internal one.

File: ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java (hunk 1 above)
Comment (cys1102) [reply ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java]: Didn't notice that internal files can be only accessed by rooted devices and emulators. If saving data in an external directory, must users have an SD card?

Comment (Kalaiz) [reply ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java]: I believe it depends on the device. If I'm not wrong, generally devices allocate a [partition of its internal memory as external storage.](https://developer.android.com/training/data-storage/app-specific#external-select-location). 

This applies to the usual emulator virtual device we use, Pixel 3a. Furthermore, our _commsgaze_  contents are stored in this "external storage". 

We should be checking for space availability, which type of external storage to use etc, but I believe we can look into it later.

File: ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java [hunk 2]
```diff
@@ -27,13 +28,20 @@ public RealTimeTest() {
...
+            if (isInternal) {
+                FileWriter out = new FileWriter(new File(context.getFilesDir(), "testResult.txt"));
+                out.write(toString());
+                out.close();
+            }
+            else {
+                FileWriter out = new FileWriter(new File(context.getExternalFilesDir(Environment.DIRECTORY_DOCUMENTS), "testResult.txt"));
+                out.write(toString());
+                out.close();
+            }
```
Comment (cys1102) [inline ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java:43]: By @Kalaiz's suggestion, I added the option for external storage.

File: ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java (hunk 1 above)
Comment (cys1102) [reply ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java]: Please, review the new commit I've done below.

File: ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java (hunk 2 above)
Comment (Kalaiz) [reply ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java:43]: Looks good!

File: ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java (hunk 1 above)
Comment (Kalaiz) [reply ProjectSpace/explore/src/main/java/com/pwc/explore/RealTimeTest.java]: Look good!

**Review by Kalaiz — APPROVED** [review]

**Pull request merged**